gensim/models/word2vec.py

```diff
@@ -1158,18 +1171,37 @@ def scan_vocab(self, sentences, progress_per=10000, trim_rule=None):
                     )
                 checked_string_types += 1
             if sentence_no % progress_per == 0:
+
                 logger.info(
                     "PROGRESS: at sentence #%i, processed %i words, keeping %i word types",
-                    sentence_no, total_words, len(vocab)
+                    sentence_no, total_words, len(vocab) if bounter_size is None else word_counts.cardinality()
                 )
-            for word in sentence:
-                vocab[word] += 1
+
             total_words += len(sentence)
 
+            if bounter_size is not None:
+                word_counts.update(sentence)
+            else:
+                for word in sentence:
+                    vocab[word] += 1
+
             if self.max_vocab_size and len(vocab) > self.max_vocab_size:
+                if bounter_size is not None:
+                    vocab = dict(word_counts)
```

Comment:
People are unlikely to use both `max_vocab_size` & a bounded-counter... but *if* they did, this branch could be hit many times during a scan, forcing multiple conversions-back-to-a-normal-dict. And after being hit once, what does it mean that both 'word_counts' and 'vocab' exist, with different sets of words? And then a few lines later, any pruned `vocab` is clobbered with another `word_counts`-conversion. So this branch is currently nonsensical.